### PR TITLE
Add rotary demos

### PR DESCRIPTION
**Pull request by Kpeved**
#### WHAT
Demos for rotary modifiers were added

#### WHY
Now it's easier to see these modifiers in action

#### HOW
A couple of screens with different modifiers

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 1]
```diff
@@ -104,6 +104,19 @@ public value class RotaryHapticsType(private val type: Int) {
...
+/**
+ * Remember disabled haptics
+ */
+@ExperimentalHorologistComposeLayoutApi
+@Composable
+public fun rememberDisabledHaptic(): RotaryHapticFeedback = remember {
+    object : RotaryHapticFeedback {
+        @ExperimentalHorologistComposeLayoutApi
+        override fun performHapticFeedback(type: RotaryHapticsType) {
+        }
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:116]: nit: leave a comment to make explicit that it was intentional to not take any action

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 2]
```diff
@@ -104,6 +104,19 @@ public value class RotaryHapticsType(private val type: Int) {
...
 }
 
+/**
+ * Remember disabled haptics
+ */
+@ExperimentalHorologistComposeLayoutApi
+@Composable
+public fun rememberDisabledHaptic(): RotaryHapticFeedback = remember {
+    object : RotaryHapticFeedback {
+        @ExperimentalHorologistComposeLayoutApi
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:114]: is this annotation needed?

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt [hunk 3]
```diff
@@ -135,6 +143,70 @@ public fun Modifier.rotaryWithScroll(
...
+ * }
+ * ```
+ * @param focusRequester Requests the focus for rotary input
+ * @param rotaryScrollAdapter A connection between scrollable objects and rotary events
+ * @param rotaryHaptics Class which will handle haptic feedback
+ */
+@ExperimentalHorologistComposeLayoutApi
+@Suppress("ComposableModifierFactory")
+@Composable
+public fun Modifier.rotaryWithSnap(
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt:165]: "add rotary demos" is probably burying the headline. This looks substantially new.

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 4]
```diff
@@ -104,6 +104,19 @@ public value class RotaryHapticsType(private val type: Int) {
     }
 }
 
+/**
+ * Remember disabled haptics
+ */
+@ExperimentalHorologistComposeLayoutApi
+@Composable
+public fun rememberDisabledHaptic(): RotaryHapticFeedback = remember {
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:112]: Nice!

**Review by yschimke — APPROVED** [review]
~Suggest updating the title~

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt (hunk 3 above)
Comment (Kpeved) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt:165]: Looks like I created this PR on top of the previous one.. Let me rebase it

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+            modifier = Modifier.fillMaxWidth(),
+            onClick = onItemClicked,
+            label = { Text("List item $it") },
+            colors = ChipDefaults.secondaryChipColors(),
+            icon = {
+                Box(
+                    modifier = Modifier
+                        .size(15.dp)
+                        .clip(CircleShape)
+                        .background(color = colors[it % 4])
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt:345]: nit: `colors[it % colors.size]`, same in the function above

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+    val items = arrayListOf(
+        "Small items",
+        "Medium cards",
+        "Very big cards",
+        "Different cards",
+        "10 small 1 big"
+    )
+    val randomHeights: List<Int> = remember { (0..300).map { Random.nextInt(1, 10) } }
+    val tenSmallOneBig: List<Int> = remember { (0..4).map { 1 }.plus(20).plus((0..4).map { 1 }) }
+    if (showList) {
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt:180]: nit: this composable function is big, could be broken into two small ones, suggesting here in this `if` statement

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+) {
+    SectionedList(
+        focusRequester = focusRequester,
+        scalingLazyListState = scalingLazyListState,
+        modifier = modifier
+    ) {
+        section(
+            listOf(
+                Pair(
+                    "Scroll list",
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt:81]: nit: these strings should be in the strings resource file

**Review by luizgrp — APPROVED** [review]
Thanks, these demos are going to be very helpful!!

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt (hunk 2 above)
Comment (Kpeved) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:114]: Probably not, removed

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt (hunk 5 above)
Comment (Kpeved) [reply sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt:345]: Done

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt (hunk 6 above)
Comment (Kpeved) [reply sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt:177]: Moved settings aside

File: sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt (hunk 7 above)
Comment (Kpeved) [reply sample/src/main/java/com/google/android/horologist/rotary/RotaryScrollMenuList.kt]: Moved to strings.xml

**Pull request merged**